### ISSUE TITLE
Please make radio-browser.info LibreJS compatible

### ISSUE THREAD

**Issue by davidhedlund** [segler-alex/radiobrowser#72]
I will make sure radio-browser.info is listed the Free Software Directory if you can make it fully compatible with LibreJS.

![screenshot from 2018-02-12 15 42 36](https://user-images.githubusercontent.com/6416073/36102101-7d64df98-100b-11e8-9f7d-5c3b78c85106.png)

**Comment by davidhedlund** [comment]
@segler-alex This is the only issue that I really care about for 100% right now. Nonfree JS is taken serious in the free software community since a few years back. I suggest that you install LibreJS and read the error messages it displays for radio-browser.info.

You said that you are rewriting this project in Rust, this is a golden opportunity to get rid of nonfree JS.

What do you think what I just said?

**Comment by segler-alex** [comment]
what is non free javascript? i did not know what librejs is. so i looked at the homepage. i tried the firefox plugin they offer. it would not install on newest firefox. i still  don't know :( could you poing me in the right direction?

**Comment by davidhedlund** [comment]
Of course. First of all, let me apology again for all bugs and suggestions an spamming that I'm responsible for occasionally. I'm almost obsessed i your projects because I like them so much so I it's been hard for me to stay away from this.

Nonfree JS is simply JS not distributed under a [free software license](https://www.gnu.org/licenses/license-list.html#SoftwareLicenses).

https://addons.mozilla.org/en-US/firefox/addon/librejs/ says:
> For explanation of the issue, see [The JavaScript Trap](https://www.gnu.org/philosophy/javascript-trap.html).

LibreJS in inbuilt in [GNU IceCat](https://www.gnu.org/software/gnuzilla/). IceCat is a freedom focused browser based on Firefox ESR. Please read the [IceCat](https://directory.fsf.org/wiki/IceCat) Free Software Directory page that I maintain to understand the differences between Firefox ESR and IceCat and why this project is important.

See also [Make your site LibreJS-friendly and HTML5-valid with rel=jslicense attribute](https://www.fsf.org/blogs/licensing/rel-jslicense)

**Comment by davidhedlund** [comment]
Please try to install [LibreJS](https://addons.mozilla.org/en-US/firefox/addon/librejs/) from addons.mozilla.org.

**Comment by davidhedlund** [comment]
Also, LibreJS is a an add-on (aka extension), not a plugin. Plugins are not widely used any more due to their negative impact on browser security. I suggest that you replace "plugin" with "add-on" in you text in https://github.com/segler-alex/radiobrowser/issues/72#issuecomment-366758907 -- I'll delete this message if you do that.

**Comment by davidhedlund** [comment]
@segler-alex The [official web page](https://www.gnu.org/software/librejs/) for LibreJS mention this
> If you administer a website, please read instructions on how to [set your JavaScript free](https://www.gnu.org/software/librejs/free-your-javascript.html) so that it respects your users' freedom. This is also documented in [this section](https://www.gnu.org/software/librejs/manual/html_node/Setting-Your-JavaScript-Free.html) of the LibreJS manual.

**Comment by davidhedlund** [comment]
I strongly recommend that you don't use any JS at all to list the radio stations at all to make the site compatible with text base radio browsers, and that you use free JS for the media player that's inbuilt in radio-browser.info.

**Comment by segler-alex** [comment]
it is not possible to not use JS. i split the program in to parts:
* the server part: only give a public API who anybody can use
* a client part: html+js that does check the public API
you need at least some kind of program that does check the API, and JS can do that.
i do not want to go back to that bad PHP all in one solution, that did not nead JS but also was a mess and hard to manage.

**Comment by davidhedlund** [comment]
> it is not possible to not use JS. i split the program in to parts:
> 
>     the server part: only give a public API who anybody can use
>     a client part: html+js that does check the public API
> 
> you need at least some kind of program that does check the API, and JS can do that.
> i do not want to go back to that bad PHP all in one solution, that did not nead JS but also was a mess and hard to manage.

Then it is possible to do without JS but not convenient with PHP. I thought you were going to rewrite the site in Rust, is it convenient enough to solve this issue?

**Comment by davidhedlund** [comment]
@segler-alex Please have a look at this issue again.

> it is not possible to not use JS. i split the program in to parts:
> 
>     * the server part: only give a public API who anybody can use
> 
>     * a client part: html+js that does check the public API
> 
> 
> you need at least some kind of program that does check the API, and JS can do that.
> i do not want to go back to that bad PHP all in one solution, that did not nead JS but also was a mess and hard to manage.

Then it is possible to do without JS but not convenient with PHP. I thought you were going to rewrite the site in Rust, is it convenient enough to solve this issue?
